api-server/src/server/component-passport.js

```diff
@@ -118,10 +118,29 @@ export const createPassportCallbackAuthenticator =
           return next(err);
         }
 
-        if (!user || !userInfo) {
+        if (!user && !userInfo) {
           return res.redirect('/signin');
         }
 
+        const state = req && req.query && req.query.state;
+        // returnTo, origin and pathPrefix are audited by getReturnTo
+        let { returnTo, origin, pathPrefix } = getReturnTo(state, jwtSecret);
+        const redirectBase = getPrefixedLandingPath(origin, pathPrefix);
+
+        if (!user && userInfo == 'access_denied') {
+          const { error_description } = req.query;
+          const blockedByLaw =
+            error_description === 'Access denied from your location';
```

Comment:
Something to think about: Are we positive these description values will not change in patches/minors?

I would not have thought the only method provided by this API to perform this sort of comparison would be a case-sensitive string.

I have put into my todo list to investigate this API.

Reply:
That's a good question & fair concern.

This string 'Access denied from your location' is a default that is set by one of the add-ons we are going to be using to enforce the restriction: https://marketplace.auth0.com/integrations/country-based-access?tab=install

In theory, this can be changed, and this check will fail, but we control "who" changes the string.

Eventually, I do plan to make the "error_description" value to a more unique/non-human friendly value, say: "us_export_regulation", but I am keeping it as is for now because we do not yet know how to use this correctly outside of "/learn", for example in forum workflows, etc. 

At least there the logs, etc. would show better info to a normal user.